### ISSUE TITLE
Error compiling Tora in linux Mint

### ISSUE THREAD

**Issue by jorgerdc** [tora-tool/tora#92]
Hi,
I'm trying to compile Tora 3.2 in Linux Mint 18.3  (Sylvia)  and I got the following error after 77%
``` 
**[ 77%] Building CXX object src/CMakeFiles/tora.dir/editor/tosqltext.cpp.o
/home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.cpp: In constructor ‘toSqlText::toSqlText(QWidget*, const char*)’:
/home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.cpp:74:38: error: invalid use of incomplete type ‘class QAction’
     , m_wrap(new QAction("Wrap", this))
                                      ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qframe.h:37:0,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qabstractscrollarea.h:37,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QAbstractScrollArea:1,
                 from /usr/include/x86_64-linux-gnu/qt5/Qsci/qsciscintillabase.h:30,
                 from /usr/include/x86_64-linux-gnu/qt5/Qsci/qscicommand.h:31,
                 from /usr/include/x86_64-linux-gnu/qt5/Qsci/qsciscintilla.h:36,
                 from /home/jorge/Escritorio/tora-3.2/src/editor/toscintilla.h:37,
                 from /home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.h:37,
                 from /home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.cpp:35:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qwidget.h:61:7: note: forward declaration of ‘class QAction’
 class QAction;
       ^
/home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.cpp:75:42: error: invalid use of incomplete type ‘class QAction’
     , m_indent(new QAction("Indent", this))
                                          ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qframe.h:37:0,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/qabstractscrollarea.h:37,
                 from /usr/include/x86_64-linux-gnu/qt5/QtWidgets/QAbstractScrollArea:1,
                 from /usr/include/x86_64-linux-gnu/qt5/Qsci/qsciscintillabase.h:30,
                 from /usr/include/x86_64-linux-gnu/qt5/Qsci/qscicommand.h:31,
                 from /usr/include/x86_64-linux-gnu/qt5/Qsci/qsciscintilla.h:36,
                 from /home/jorge/Escritorio/tora-3.2/src/editor/toscintilla.h:37,
                 from /home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.h:37,
                 from /home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.cpp:35:
/usr/include/x86_64-linux-gnu/qt5/QtWidgets/qwidget.h:61:7: note: forward declaration of ‘class QAction’
 class QAction;
       ^
/home/jorge/Escritorio/tora-3.2/src/editor/tosqltext.cpp:113:11: error: invalid use of incomplete type ‘class QAction’
     m_wrap->setCheckable(true);
           ^**  
```
my cmake command is:
cmake -DBOOST_ROOT=/usr/include -DCMAKE_INSTALL_PREFIX:PATH=/opt/tora-3.2 -DCMAKE_BUILD_TYPE=Release ..

Trying with  version 3.1 there is no issue, but I would like to get version 3.2 installed.
Any suggestion?

**Comment by ssttaarr** [comment]
@jorgerdc 

I've succeeded to compile TOra v.3.2 on my Mint box today (Linux Mint 18.3 Sylvia, gcc-5.4.0, qt-5.5.1)

I've used an approach described in https://help.ubuntu.com/community/HowToBuildToraWithOracle to build .deb package with the following changes:
```Diff
diff --git a/cmake/modules/PrecompiledHeader.cmake b/cmake/modules/PrecompiledHeader.cmake
index 172d0d7..f06b486 100644
--- a/cmake/modules/PrecompiledHeader.cmake
+++ b/cmake/modules/PrecompiledHeader.cmake
@@ -95,7 +95,7 @@ FUNCTION(ADD_PRECOMPILED_HEADER _targetName _input)
     MESSAGE("${CMAKE_CXX_COMPILER} -DPCHCOMPILE ${_compiler_FLAGS} -x c++-header -o {_output} ${_source}")
     ADD_CUSTOM_COMMAND(
       OUTPUT ${_output}
-      COMMAND ${CMAKE_CXX_COMPILER} ${_compiler_FLAGS} -x c++-header -o ${_output} ${_source}
+      COMMAND ${CMAKE_CXX_COMPILER} ${_compiler_FLAGS} -std=c++11 -fPIC -x c++-header -o ${_output} ${_source}
       DEPENDS ${_source} )
     ADD_CUSTOM_TARGET(${_targetName}_gch DEPENDS ${_output})
     ADD_DEPENDENCIES(${_targetName} ${_targetName}_gch)
diff --git a/debian/rules b/debian/rules
index 10334db..b5c86b7 100755
--- a/debian/rules
+++ b/debian/rules
@@ -24,7 +24,7 @@ install/tora::
 	dh $@ --buildsystem=cmake
 
 override_dh_auto_configure:
-	dh_auto_configure -- -DCMAKE_BUILD_TYPE=Debug -DUSE_PCH=true
+	dh_auto_configure -- -DCMAKE_BUILD_TYPE=Debug -DUSE_PCH=true -DBOOST_ROOT=/usr/include -DORACLE_PATH_INCLUDES=/usr/include/oracle/12.2/client64
 
 override_dh_install:
 	@echo "skip dh_install"
```
Changes to the first were required to fix error related to precompiled headers. Changes to the second file are related to Boost library (described in README.CMAKE) and to Oracle include files.

Could you problem be related to Qt version?

**Comment by jorgerdc** [comment]
@ssttaarr ,
Thank you for your suggestion,  I'll try to compile using these suggestions.

**Comment by ibre5041** [comment]
Fixed by 5eccd7e28 - do not use precompiled headers.
